rules/engine.go

```diff
@@ -63,6 +64,7 @@ type V3Engine interface {
 		preconditions DynamicRule,
 		ttl int,
 		callback V3RuleTaskCallback) error
+	SetWatcherWrapper(WrapWatcher)
```

Comment:
Just to clarify, this is just used during testing with the mock, right?

Reply:
No this method is called from external projects to set the appropriate watch wrapper for the engine

Reply:
It has a similar use case and implementation as `SetKVWrapper(WrapKV)`

Reply:
ok, thx, will have to review the eternal project use for this more to better understand.